dash/include/dash/halo/Halo.h

```diff
@@ -281,137 +277,141 @@ std::ostream& operator<<( std::ostream&                        os,
+    for(auto& spec : tmp) {
       _specs[spec.index()] = spec;
       ++_num_regions;
     }
   }
 
-  HaloSpec( const self& other ) { _specs = other._specs; }
+  HaloSpec(const Self_t& other) { _specs = other._specs; }
 
-  constexpr HaloRegionSpecT spec( const region_index_t index ) const {
+  constexpr RegionSpec_t spec(const region_index_t index) const {
     return _specs[index];
   }
 
-  constexpr extent_t extent( const region_index_t index ) const {
+  constexpr region_extent_t extent(const region_index_t index) const {
     return _specs[index].extent();
   }
 
-  constexpr std::size_t numRegions() const { return _num_regions; }
+  constexpr region_size_t num_regions() const { return _num_regions; }
 
-  const SpecsT& haloSpecs() const { return _specs; }
+  const Specs_t& specs() const { return _specs; }
 
- private:
-  void setRegionSpec( const StencilT& stencil ) {
-    auto index = HaloRegionSpecT::index( stencil );
+private:
+  void set_region_spec(const Stencil_t& stencil) {
+    auto index = RegionSpec_t::index(stencil);
     auto max   = stencil.max();
 
-    if( _specs[index].extent() == 0 )
+    if(_specs[index].extent() == 0)
       ++_num_regions;
 
-    if( max > _specs[index].extent() )
-      _specs[index] = HaloRegionSpecT( index, max );
+    if(max > _specs[index].extent())
+      _specs[index] = RegionSpec_t(index, max);
   }
 
-  bool nextRegion( const StencilT& stencil, StencilT& stencil_combination ) {
-    for( auto d = 0; d < NumDimensions; ++d ) {
-      if( stencil[d] == 0 )
+  bool next_region(const Stencil_t& stencil, Stencil_t& stencil_combination) {
+    for(auto d = 0; d < NumDimensions; ++d) {
+      if(stencil[d] == 0)
         continue;
-      stencil_combination[d] = ( stencil_combination[d] == 0 ) ? stencil[d] : 0;
-      if( stencil_combination[d] == 0 ) {
+      stencil_combination[d] = (stencil_combination[d] == 0) ? stencil[d] : 0;
+      if(stencil_combination[d] == 0) {
         return true;
       }
     }
 
     return false;
   }
 
- private:
-  SpecsT      _specs{};
-  std::size_t _num_regions{ 0 };
+private:
+  Specs_t       _specs{};
+  region_size_t _num_regions{ 0 };
 };  // HaloSpec
 
 template <typename ElementT, typename PatternT,
           typename PointerT   = GlobPtr<ElementT, PatternT>,
           typename ReferenceT = GlobRef<ElementT>>
-class RegionIter
-: public std::iterator<std::random_access_iterator_tag, ElementT,
-                       typename PatternT::index_type, PointerT, ReferenceT> {
- private:
-  using SelfT = RegionIter<ElementT, PatternT, PointerT, ReferenceT>;
-  using GlobMemT =
+class RegionIter {
+private:
+  using Self_t = RegionIter<ElementT, PatternT, PointerT, ReferenceT>;
+  using GlobMem_t =
     GlobStaticMem<ElementT, dash::allocator::SymmetricAllocator<ElementT>>;
-  using ViewSpecT = typename PatternT::viewspec_type;
+  using ViewSpec_t      = typename PatternT::viewspec_type;
+  using pattern_index_t = typename PatternT::index_type;
```

Comment:
why `pattern_index_t` and not just `index_t`? Is there another (conflicting) index type?

Reply:
The thing is, that if i use index_t, i also have to use size_t. Then it might be not clear whether it is std::size_t or an other. it's an ugly name clash. i also use region_size_t.

Reply:
Gnarf, how about `size_type` and `index_type`?

Reply:
Personally i hate type as ending. Typically _t is used to indicate a type. Especially longer names don't profit from type.